pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdFunctions.java

```diff
@@ -323,56 +327,129 @@ void processArguments() throws Exception {
                 throw new IllegalArgumentException("Effectively-once processing semantics can only be achieved using a Failover subscription type");
             }
 
-            functionConfig.setAutoAck(true);
+            // window configs
+            WindowConfig windowConfig = functionConfig.getWindowConfig();
+            if (null != windowLengthCount) {
+                if (windowConfig == null) {
```

Comment:
This `if (windowConfig == null)` logic could be a separate function to make things a bit more DRY (note the repetition below.

Reply:
I am not sure that buys us much.  I can make the if statement one line:

if (windowConfig == null) windowConfig = new WindowConfig()

but having another function like this:

  private void setWindowConfig(Object config, WindowConfig windowConfig) {
            if (config != null) {
                windowConfig = new WindowConfig();
            }
        }

is an overkill I think